amino.go

```diff
@@ -340,6 +371,26 @@ func (cdc *Codec) UnmarshalBinaryBare(bz []byte, ptr interface{}) error {
 		}
 		bz = bz[4:]
 	}
+	isRepeatedStruct := info.Type.Kind() == reflect.Slice && info.Type.Elem().Kind() == reflect.Struct
```

Comment:
Maybe we ought to have an auxiliary function given an `info` that returns a bool if it's a repeated struct.

Reply:
Changed. Also, added another test case and a few related changes in 8764911